File: app/src/main/java/fathi/shakhes/MainActivity.java
package fathi.shakhes;

import android.content.Intent;
import android.graphics.Color;
import android.graphics.Typeface;
import android.os.Bundle;
import android.os.Handler;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import shakhes.R;

public class MainActivity extends AppCompatActivity {
    boolean BackPress = false;

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = findViewById(R.id.toolbar4);
        setSupportActionBar(toolbar);
        getSupportActionBar().setLogo(R.drawable.shariflogo);
        getSupportActionBar().setDisplayOptions(ActionBar.DISPLAY_SHOW_CUSTOM);

        View v = LayoutInflater
                .from(getSupportActionBar().getThemedContext())
                .inflate(R.layout.action_bar1, null);
        ActionBar.LayoutParams params = new ActionBar.LayoutParams(
                ActionBar.LayoutParams.MATCH_PARENT,
                ActionBar.LayoutParams.MATCH_PARENT);
        getSupportActionBar().setCustomView(v, params);

        Typeface typeFace = Typeface.createFromAsset(getAssets(), "fonts/IRANSans.ttf");
        TextView myTextView1 = findViewById(R.id.text1);
        TextView myTextView2 = findViewById(R.id.text2);
        myTextView1.setTypeface(typeFace);
        myTextView2.setTypeface(typeFace);
    }

    public void food(View view) {
        final Handler handler = new Handler();
        handler.postDelayed(() -> {
            Intent FoodIntent = new Intent(MainActivity.this, LoginFood.class);
            startActivity(FoodIntent);
        }, 270);
    }

    @Override
    public void onBackPressed() {
        if (BackPress) {
            super.onBackPressed();
            overridePendingTransition(0, R.anim.fade_out);
            MainActivity.this.finish();
        }
        this.BackPress = true;
        Toast.makeText(getApplicationContext(), "برای خروج دو بار کلیک کنید", Toast.LENGTH_SHORT).show();
        new Handler().postDelayed(() -> BackPress = false, 2000);
    }

}

File: app/src/main/java/fathi/shakhes/MainActivity.kt
package fathi.shakhes

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.FragmentManager.POP_BACK_STACK_INCLUSIVE
import androidx.lifecycle.lifecycleScope
import fathi.shakhes.base.openFragment
import fathi.shakhes.fragments.LoginFragment
import fathi.shakhes.fragments.MainFragment
import fathi.shakhes.fragments.SplashFragment
import fathi.shakhes.helpers.showExitAppDialog
import kotlinx.coroutines.delay
import shakhes.R

class MainActivity : AppCompatActivity() {
    init {
        lifecycleScope.launchWhenCreated {
            delay(2000)
            openNextFragment()
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        setTheme(R.style.AppTheme)
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        openFragment(SplashFragment(), true)
    }

    private fun openNextFragment() {
        supportFragmentManager.popBackStack(null, POP_BACK_STACK_INCLUSIVE)
        openFragment(
            if (shouldOpenLoginFragment()) LoginFragment() else MainFragment(), false
        )
    }

    private fun shouldOpenLoginFragment(): Boolean = !AppSharedPreferences.shouldSaveLoginData
            || AppSharedPreferences.accessToken.isBlank()

    override fun onNewIntent(intent: Intent?) {
        super.onNewIntent(intent)
        openNextFragment()
    }

    override fun onBackPressed() {
        val count = supportFragmentManager.backStackEntryCount
        if (count == 0) {
            showExitAppDialog(this)
        } else {
            supportFragmentManager.popBackStack()
        }
    }

}